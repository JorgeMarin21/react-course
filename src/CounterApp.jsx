import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ( { value } ) => {
  if ( !value && value !== 0 )
  throw new Error( 'No existe valor');

  const [ counter, setCounter ] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1 )
    // setCounter( (c) => c + 1 )
  }
  const handleSubstractAdd = () => {
    setCounter(counter - 1 )
    // setCounter( (c) => c - 1 )
  }
  const handleReset = () => {
    setCounter( value )
    // setCounter( (c) => c - 1 )
  }

  return (
    <>
      <h1> CounterApp </h1>
      <h2> { counter } </h2>
      <button aria-label="btn-plus" onClick={ handleAdd }>+1</button>
      <button aria-label="btn-minus" onClick={ handleSubstractAdd }>-1</button>
      <button aria-label="btn-reset" onClick={ handleReset }>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}