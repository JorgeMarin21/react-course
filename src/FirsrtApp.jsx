import PropTypes from 'prop-types'
const getMessage = () => 'Hola Mijo'

export const FirstApp = ( { title, subtitle = 'vamos' } ) => {

  if ( !title )
  throw new Error( 'prueba')
  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      <p> { subtitle } </p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

// FirstApp.defaultProps = {
//   title: 'No hay titulo'
// }
// defaultProps Deprecated