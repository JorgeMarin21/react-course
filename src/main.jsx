import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorldApp';
// import { CounterApp } from './CounterApp';
import './styles.css';
// ReactDOM.createRoot(document.getElementById('root')).render(
//     <React.StrictMode>
//         <FirstApp title = 'El titulo' subtitle = 'El subtitulo' />
//     </React.StrictMode>
// );
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value = { 0 } />
    </React.StrictMode>
);