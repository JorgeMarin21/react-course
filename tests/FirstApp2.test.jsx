import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirsrtApp";

describe('Pruebas en FirstApp', () => { 
    
    const title = 'Hi, I am a tester';
    const subtitle = 'and What is your name?';

    test('should match with snapshot', () => { 
        const { container } = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
     });

    test('should show message "Hi, I am tester"', () => { 
        render(<FirstApp title = {title} />);
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('should show title in a H1', () => { 
        render(<FirstApp title = {title} />);
        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain(title);
    });

    test('should show subtitle send by props', () => { 
        render(<FirstApp title = {title} subtitle = {subtitle}/>);
        expect( screen.getAllByText(subtitle).length).toBe(1);
    });
 })