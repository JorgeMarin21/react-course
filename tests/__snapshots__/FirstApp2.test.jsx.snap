// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hi, I am a tester
  </h1>
  <p>
     
    vamos
     
  </p>
</div>
`;
